### PR TITLE
ci: Fix Fedora Rawhide build

### PR DESCRIPTION
**Pull request by rst0git**
When `sudo -E` is used to run Podman it results in the `XDG_RUNTIME_DIR` environment variable being set incorrectly that prevents Podman from running.

This PR also resolves include conflicts with glibc 2.36:
https://sourceware.org/glibc/wiki/Release/2.36#Usage_of_.3Clinux.2Fmount.h.3E_and_.3Csys.2Fmount.h.3E

- `criu/include/linux/mount.h` has been updated to match `/usr/include/sys/mount.h`
- inline definitions `sys_*()` functions have been moved from "linux/mount.h" to "syscall.h" to avoid conflicts with `uapi/compel/plugins/std/syscall.h` and `<unistd.h>`.
- include of `<linux/aio_abi.h>` has been replaced with local include to avoid conflicts with `<sys/mount.h>`.

Fixes: #1942
Fixes: #1949

### CODE REVIEW

File: .github/workflows/fedora-rawhide-test.yml [hunk 1]
```diff
@@ -9,4 +9,6 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - name: Run Fedora Rawhide Test
-      run: sudo -E make -C scripts/ci fedora-rawhide CONTAINER_RUNTIME=podman BUILD_OPTIONS="--security-opt seccomp=unconfined"
+      # Make sure that XDG_RUNTIME_DIR environment variable is not set.
+      # https://github.com/checkpoint-restore/criu/issues/1942
```
Comment (avagin) [inline .github/workflows/fedora-rawhide-test.yml:13]: pls don't add issue in the code if it isn't todo/fixme. The comment has to contain all required info to understand the code.

File: .github/workflows/fedora-rawhide-test.yml [hunk 2]
```diff
@@ -9,4 +9,6 @@ jobs:
...
+      run: sudo -E XDG_RUNTIME_DIR= make -C scripts/ci fedora-rawhide CONTAINER_RUNTIME=podman BUILD_OPTIONS="--security-opt seccomp=unconfined"
```
Comment (avagin) [inline .github/workflows/fedora-rawhide-test.yml:14]: why do we set -E in this case? Do we pass any env var-s?

**Review by adrianreber — APPROVED** [review]
I have seen this also mentioned as a possible solution. Looks good to me.

File: .github/workflows/fedora-rawhide-test.yml (hunk 2 above)
Comment (adrianreber) [reply .github/workflows/fedora-rawhide-test.yml:14]: We need environment variables from the CI environment to distinguish between CI environments.

File: .github/workflows/fedora-rawhide-test.yml (hunk 1 above)
Comment (rst0git) [reply .github/workflows/fedora-rawhide-test.yml]: Thanks Andrei, I have updated the comment with FIXME for the GitHub issue in Podman.

**Comment by mihalicyn** [comment]
LGTM

**Review by mihalicyn — APPROVED** [review]

**Pull request merged**